scheduler/scheduler.go

```diff
@@ -139,22 +146,87 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 			continue
 		}
 
+		if ctx.Err() != nil {
+			return nil, errors.New("scheduling run canceled.")
+		}
+
+		d, err := distro.FindOne(distro.ById(distroId))
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"distro":  distroId,
+				"runner":  RunnerName,
+				"message": "failed to find distro",
+			}))
+			continue
+		}
+
+		CurrentParents, err := host.FindAllRunningParents()
+		if err != nil {
+			return nil, err
+		}
+		ExistingContainers, err := host.FindAllRunningContainers()
+		if err != nil {
+			return nil, err
+		}
+
 		hostsSpawnedPerDistro[distroId] = make([]host.Host, 0, numHostsToSpawn)
-		for i := 0; i < numHostsToSpawn; i++ {
-			if ctx.Err() != nil {
-				return nil, errors.New("scheduling run canceled.")
+
+		// if distro can have containers, check if there are enough parents to hold
+		// new containers
+		if d.MaxContainers != 0 {
```

Comment:
`> 0`, in case we miss some validation somewhere, and someone sets negative `MaxContainers`.